activestorage/lib/active_storage/attached/changes/create_one.rb

```diff
@@ -68,6 +68,18 @@ def find_or_build_blob
               service_name: attachment_service_name
             ).symbolize_keys
           )
+        when URI
+          filename = File.basename(attachable.path)
+          io = attachable.open
+          content_type = Marcel::MimeType.for(io, name: filename)
```

Comment:
I think that's already happening automatically here: https://github.com/rails/rails/blob/6287c109d3362985d352160fc0fe674ed899d6d2/activestorage/app/models/active_storage/blob.rb#L249
https://github.com/rails/rails/blob/6287c109d3362985d352160fc0fe674ed899d6d2/activestorage/app/models/active_storage/blob.rb#L368-L370